processing/src/main/java/io/druid/segment/DimensionHandler.java

```diff
@@ -0,0 +1,181 @@
+   * and building the on-disk representation of a dimension. For use with IndexMergerV9 only.
+   *
+   * See {@link DimensionMergerV9} interface for more information.
+   *
+   * @param indexSpec     Specification object for the index merge
+   * @param outDir        Location to store files generated by the merging process
+   * @param ioPeon        ioPeon object passed in by IndexMerger, manages files created by the merging process
+   * @param capabilities  The ColumnCapabilities of the dimension represented by this DimensionHandler
+   * @param progress      ProgressIndicator used by the merging process
+
+   * @return A new DimensionMergerV9 object.
+   */
+  public DimensionMergerV9<EncodedTypeArray> makeMerger(
+      IndexSpec indexSpec,
+      File outDir,
+      IOPeon ioPeon,
+      ColumnCapabilities capabilities,
+      ProgressIndicator progress
+  );
+
+
+  /**
+   * Creates a new DimensionMergerLegacy, a per-dimension object responsible for merging indexes/row data across segments
+   * and building the on-disk representation of a dimension. For use with IndexMerger only.
+   *
+   * See {@link DimensionMergerLegacy} interface for more information.
+   *
+   * @param indexSpec     Specification object for the index merge
+   * @param outDir        Location to store files generated by the merging process
+   * @param ioPeon        ioPeon object passed in by IndexMerger, manages files created by the merging process
+   * @param capabilities  The ColumnCapabilities of the dimension represented by this DimensionHandler
+   * @param progress      ProgressIndicator used by the merging process
+
+   * @return A new DimensionMergerLegacy object.
+   */
+  public DimensionMergerLegacy<EncodedTypeArray> makeLegacyMerger(
+      IndexSpec indexSpec,
+      File outDir,
+      IOPeon ioPeon,
+      ColumnCapabilities capabilities,
+      ProgressIndicator progress
+  );
+
+
+  /**
+   * Given an array representing a single set of row value(s) for this dimension as an Object,
+   * return the length of the array after appropriate type-casting.
+   *
+   * For example, a dictionary encoded String dimension would receive an int[] as an Object.
+   *
+   * @param dimVals Array of row values
+   * @return Size of dimVals
+   */
+  public int getLengthFromEncodedArray(EncodedTypeArray dimVals);
+
+
+  /**
+   * Given two arrays representing sorted encoded row value(s), return the result of their comparison.
+   *
+   * If the two arrays have different lengths, the shorter array should be ordered first in the comparison.
+   *
+   * Otherwise, this function should iterate through the array values and return the comparison of the first difference.
+   *
+   * @param lhs array of row values
+   * @param rhs array of row values
+   * @return integer indicating comparison result of arrays
+   */
+  public int compareSortedEncodedArrays(EncodedTypeArray lhs, EncodedTypeArray rhs);
```

Comment:
It's used only in the Rowboat compareTo() function, it doesn't depend on any merger state so I put it on the DimensionHandler, I can move it if needed